thunder/torch/__init__.py

```diff
@@ -5064,6 +5066,28 @@ def all_gather(
 
         return dist_prims.all_gather(a, group, async_op, dim=dim)
 
+    @torchsymbol(
+        torch.distributed.all_gather_into_tensor,
+        is_method=False,
+        id="all_gather_",
+        tags=(prims.OpTags.IN_PLACE,),
+    )
+    def all_gather_(
+        output_tensor: TensorLike,
+        input_tensor: TensorLike,
+        /,
+        group: torch.distributed.ProcessGroup | None = None,
+        async_op: bool = False,
+    ) -> TensorLike:
+        utils.check(
+            not async_op,
+            lambda: f"`torch.distributed.all_gather_into_tensor` with {async_op=} is not supported",
+            NotImplementedError,
+        )
+        group = group if group is not None else torch.distributed.new_group()
+        out = dist_prims.all_gather(input_tensor, group, async_op, dim=None, output_tensor=output_tensor)
+        return prims.copy_(out, output_tensor)
```

Comment:
What do you think about also using `wait` in this decomposition? Is this additional information in the trace useful?
```suggestion
        async_op = True
        out = dist_prims.all_gather(input_tensor, group, async_op, dim=None, output_tensor=output_tensor)
        out = dist_prims.wait(out)
        return prims.copy_(out, output_tensor)
```